### PR TITLE
Image refresh for rhel-9-0

### PR DESCRIPTION
**Pull request by cockpituous**
Image refresh for rhel-9-0
 * [x] image-refresh rhel-9-0
 * [x] Adjust check-connection for OpenSSL 3: https://github.com/cockpit-project/cockpit/pull/16014
 * [ ] report and naughty new [selinux nft issue](https://logs.cockpit-project.org/logs/pull-16014-20210629-082235-d624cd34-rhel-9-0-bots-2144/log.html#66)

### CODE REVIEW

**Comment by martinpitt** [comment]
There is an [internal libvirt error in c-machines](https://logs.cockpit-project.org/logs/pull-2144-20210625-023043-aaa593e3-rhel-9-0-cockpit-project-cockpit-machines/log.html#43-2) which needs to be reported and naughtied. But in general the SELinux thing that broke all of libvirt seems fixed.

**Comment by martinpitt** [comment]
https://github.com/cockpit-project/cockpit-machines/pull/223 landed, retrying c-machines. Eventually we should rebase this PR to take other new image refreshes into account, and reset test fail statistics, but it's nice to see the progress.

**Comment by martinpitt** [comment]
[This SELinux failure](https://logs.cockpit-project.org/logs/pull-2144-20210628-104836-aaa593e3-rhel-9-0-cockpit-project-cockpit/log.html#44) is already tracked in known issue #2141 , we just need to copy the naughty. @KKoukiou FYI.

I didn't look at the testTls cockpit failure yet, but can do today (I suppose that's more my area of expertise than Katerina's)

**Comment by martinpitt** [comment]
RHEL 9 landed OpenSSL 3.0, which is surely related to the [cockpit testTls failure](https://logs.cockpit-project.org/logs/pull-2144-20210628-104836-aaa593e3-rhel-9-0-cockpit-project-cockpit/log.html#23-2). In particular, the CLI can't read files from stdin any more:
```
# openssl x509 -in - -text < /usr/share/gnupg/sks-keyservers.netCA.pem
Could not open file or uri for loading certificate from -
4027E1A4227F0000:error:16000069:STORE routines:ossl_store_get0_loader_int:unregistered scheme:crypto/store/store_register.c:237:scheme=file
4027E1A4227F0000:error:80000002:system library:file_open:No such file or directory:providers/implementations/storemgmt/file_store.c:266:calling stat(-)
Unable to load certificate
```
Admittedly, the manpage says to leave out the option to read from stin -- it only happened to work until now. I'll send a PR.

After fixing this there is more fallout, I'll include that in the PR.

**Comment by martinpitt** [comment]
I sent https://github.com/cockpit-project/cockpit/pull/16014 to fix check-connection (running test against this image there), and added the naughty here. c-machines was green, cockpit gets tested in the mentioned PR, re-triggering podman to get a current result.

**Comment by KKoukiou** [comment]
Attempting to fix the podman issue on https://github.com/cockpit-project/cockpit-podman/pull/732

**Comment by martinpitt** [comment]
In the cockpit fix PR there is another new [SELinux regression](https://logs.cockpit-project.org/logs/pull-16014-20210629-082235-d624cd34-rhel-9-0-bots-2144/log.html#66), added to description's TODO list.

**Comment by KKoukiou** [comment]
I fear that the podman failure has nothing to do with this PR http://grafana-latest-osci-infra.cloud.privileged.psi.redhat.com/d/yQYsRilMz/cockpit-ci?orgId=1&var-cockpit_project=cockpit-project%2Fcockpit-podman 

It's just a very flaky test.

**Comment by KKoukiou** [comment]
@martinpitt I see you put into the description ' report and naughty new selinux nft issue' but in last run I don't see this failure. Is it something that happens from time to time?

File: naughty/rhel-9/2144-podman-api-call-loses-data [hunk 1]
```diff
@@ -0,0 +1,6 @@
+Traceback (most recent call last):
```
Comment (martinpitt) [inline naughty/rhel-9/2144-podman-api-call-loses-data:1]: wrong file name -- you want 2158-*

**Review by martinpitt — COMMENTED** [review]
Do you want to include a naughty for the nft selinux issue as well, or want me to?

**Comment by cockpituous** [comment]
rhel-9-0
Ooops, it happened again<hr>
```
# ----------------------------------------------------------------------
# testDownloadImage (__main__.TestApplication)
Warning: Stopping podman.service, but it can still be activated by:
  podman.socket
mkdir: cannot create directory '/home/admin/.ssh': File exists
Warning: Stopping podman.service, but it can still be activated by:
  podman.socket
Getting image source signatures
Copying blob sha256:d2421964bad195c959ba147ad21626ccddc73a4f2638664ad1c07bd9df48a675
Copying config sha256:f0b02e9d092d905d0d87a8455a1ae3e9bb47b4aa3dc125125ca5cd10d6441c9f
Writing manifest to image destination
Storing signatures
Getting image source signatures
Copying blob sha256:d2421964bad195c959ba147ad21626ccddc73a4f2638664ad1c07bd9df48a675
Copying config sha256:f0b02e9d092d905d0d87a8455a1ae3e9bb47b4aa3dc125125ca5cd10d6441c9f
Writing manifest to image destination
Storing signatures
[0630/090306.676111:ERROR:bus.cc(393)] Failed to connect to the bus: Failed to connect to socket /run/dbus/system_bus_socket: No such file or directory
[0630/090306.677563:WARNING:headless_browser_main_parts.cc(83)] Cannot create Pref Service with no user data dir.

DevTools listening on ws://127.0.0.1:10108/devtools/browser/fb361371-e188-43c4-860a-9fa7c3810aa2
[0630/090306.687187:ERROR:gpu_init.cc(426)] Passthrough is not supported, GL is egl
CDP: {"source":"security","level":"error","text":"Refused to apply inline style because it violates the following Content Security Policy directive: \"default-src 'self' http://127.0.0.2:9091\". Either the 'unsafe-inline' keyword, a hash ('sha256-BMIkowAqVEI88ibUJ1NtYB8ByZOuS4MTi73eGoFo2+Q='), or a nonce ('nonce-...') is required to enable inline execution. Note also that 'style-src' was not explicitly set, so 'default-src' is used as a fallback.\n","timestamp":1625043787807.836,"url":"http://127.0.0.2:9091/cockpit/$f154b83f107dfb45e7f4b78ac45b1e96cdc829fcc1b1f7723a442eba73fdab9c/shell/index.js","lineNumber":30,"stackTrace":{"callFrames":[{"functionName":"ke","scriptId":"21","url":"http://127.0.0.2:9091/cockpit/$f154b83f107dfb45e7f4b78ac45b1e96cdc829fcc1b1f7723a442eba73fdab9c/shell/index.js","lineNumber":30,"columnNumber":35691},{"functionName":"Ze","scriptId":"21","url":"http://127.0.0.2:9091/cockpit/$f154b83f107dfb45e7f4b78ac45b1e96cdc829fcc1b1f7723a442eba73fdab9c/shell/index.js","lineNumber":30,"columnNumber":44728},{"functionName":"append","scriptId":"21","url":"http://127.0.0.2:9091/cockpit/$f154b83f107dfb45e7f4b78ac45b1e96cdc829fcc1b1f7723a442eba73fdab9c/shell/index.js","lineNumber":30,"columnNumber":46409},{"functionName":"D","scriptId":"21","url":"http://127.0.0.2:9091/cockpit/$f154b83f107dfb45e7f4b78ac45b1e96cdc829fcc1b1f7723a442eba73fdab9c/shell/index.js","lineNumber":63,"columnNumber":24932},{"functionName":"","scriptId":"21","url":"http://127.0.0.2:9091/cockpit/$f154b83f107dfb45e7f4b78ac45b1e96cdc829fcc1b1f7723a442eba73fdab9c/shell/index.js","lineNumber":63,"columnNumber":34130},{"functionName":"","scriptId":"21","url":"http://127.0.0.2:9091/cockpit/$f154b83f107dfb45e7f4b78ac45b1e96cdc829fcc1b1f7723a442eba73fdab9c/shell/index.js","lineNumber":68,"columnNumber":160544},{"functionName":"","scriptId":"21","url":"http://127.0.0.2:9091/cockpit/$f154b83f107dfb45e7f4b78ac45b1e96cdc829fcc1b1f7723a442eba73fdab9c/shell/index.js","lineNumber":68,"columnNumber":160548}]}}
> error: Modal: Specify at least one of: title, aria-label, aria-labelledby.
CDP: {"source":"network","level":"error","text":"Failed to load resource: the server responded with a status of 404 (Not Found)","timestamp":1625043788060.689,"url":"http://127.0.0.2:9091/cockpit/$f154b83f107dfb45e7f4b78ac45b1e96cdc829fcc1b1f7723a442eba73fdab9c/podman/po.js","networkRequestId":"3130.66"}
CDP: {"source":"network","level":"error","text":"Failed to load resource: the server responded with a status of 404 (Not Found)","timestamp":1625043788103.823,"url":"http://127.0.0.2:9091/cockpit/$f154b83f107dfb45e7f4b78ac45b1e96cdc829fcc1b1f7723a442eba73fdab9c/podman/po.js","networkRequestId":"3130.71"}
> warning: grep: /sys/class/dmi/id/power/autosuspend_delay_ms: Input/output error

Traceback (most recent call last):
  File "/usr/lib64/python3.9/unittest/case.py", line 550, in _callTestMethod
    method()
  File "/work/bots/make-checkout-workdir/test/check-application", line 767, in testDownloadImage
    dialog0.openDialog() \
  File "/work/bots/make-checkout-workdir/test/check-application", line 707, in expectDownloadSuccess
    b.wait_visible('#containers-images td[data-label="Name"]:contains("{0}")'.format(self.imageName))
  File "/work/bots/make-checkout-workdir/test/common/testlib.py", line 411, in wait_visible
    self._wait_present(selector)
  File "/work/bots/make-checkout-workdir/test/common/testlib.py", line 402, in _wait_present
    self.wait_js_func('ph_is_present', selector)
  File "/work/bots/make-checkout-workdir/test/common/testlib.py", line 396, in wait_js_func
    self.wait_js_cond("%s(%s)" % (func, ','.join(map(jsquote, args))))
  File "/work/bots/make-checkout-workdir/test/common/testlib.py", line 393, in wait_js_cond
    self.raise_cdp_exception("timeout\nwait_js_cond", cond, result["exceptionDetails"], trailer)
  File "/work/bots/make-checkout-workdir/test/common/testlib.py", line 187, in raise_cdp_exception
    raise Error("%s(%s): %s" % (func, arg, msg))
testlib.Error: timeout
wait_js_cond(ph_is_present("#containers-images td[data-label=\"Name\"]:contains(\"my-busybox\")")): {'exceptionId': 1, 'text': 'Uncaught (in promise) Error: condition did not become true', 'lineNumber': 238, 'columnNumber': 9, 'scriptId': '3', 'stackTrace': {'callFrames': [{'functionName': 'PhWaitCondTimeout', 'scriptId': '3', 'url': '', 'lineNumber': 237, 'columnNumber': 8}, {'functionName': '', 'scriptId': '3', 'url': '', 'lineNumber': 249, 'columnNumber': 23}]}, 'exception': {'type': 'object', 'subtype': 'error', 'className': 'PhWaitCondTimeout', 'description': 'Error: condition did not become true\n    at <anonymous>:250:24', 'objectId': '9128935592755241297.6.2'}}
cdp: {'source': 'network', 'level': 'error', 'text': 'Failed to load resource: the server responded with a status of 401 (Authentication failed)', 'timestamp': 1625043787165.48, 'url': 'http://127.0.0.2:9091/cockpit/login', 'networkRequestId': '3130.12'}
cdp: {'source': 'security', 'level': 'error', 'text': 'Refused to apply inline style because it violates the following Content Security Policy directive: "default-src \'self\' http://127.0.0.2:9091". Either the \'unsafe-inline\' keyword, a hash (\'sha256-BMIkowAqVEI88ibUJ1NtYB8ByZOuS4MTi73eGoFo2+Q=\'), or a nonce (\'nonce-...\') is required to enable inline execution. Note also that \'style-src\' was not explicitly set, so \'default-src\' is used as a fallback.\n', 'url': 'http://127.0.0.2:9091/cockpit/$f154b83f107dfb45e7f4b78ac45b1e96cdc829fcc1b1f7723a442eba73fdab9c/shell/index.js', 'lineNumber': 30, 'stackTrace': {'callFrames': [{'functionName': 'ke', 'scriptId': '21', 'url': 'http://127.0.0.2:9091/cockpit/$f154b83f107dfb45e7f4b78ac45b1e96cdc829fcc1b1f7723a442eba73fdab9c/shell/index.js', 'lineNumber': 30, 'columnNumber': 35691}, {'functionName': 'Ze', 'scriptId': '21', 'url': 'http://127.0.0.2:9091/cockpit/$f154b83f107dfb45e7f4b78ac45b1e96cdc829fcc1b1f7723a442eba73fdab9c/shell/index.js', 'lineNumber': 30, 'columnNumber': 44728}, {'functionName': 'append', 'scriptId': '21', 'url': 'http://127.0.0.2:9091/cockpit/$f154b83f107dfb45e7f4b78ac45b1e96cdc829fcc1b1f7723a442eba73fdab9c/shell/index.js', 'lineNumber': 30, 'columnNumber': 46409}, {'functionName': 'D', 'scriptId': '21', 'url': 'http://127.0.0.2:9091/cockpit/$f154b83f107dfb45e7f4b78ac45b1e96cdc829fcc1b1f7723a442eba73fdab9c/shell/index.js', 'lineNumber': 63, 'columnNumber': 24932}, {'functionName': '', 'scriptId': '21', 'url': 'http://127.0.0.2:9091/cockpit/$f154b83f107dfb45e7f4b78ac45b1e96cdc829fcc1b1f7723a442eba73fdab9c/shell/index.js', 'lineNumber': 63, 'columnNumber': 34130}, {'functionName': '', 'scriptId': '21', 'url': 'http://127.0.0.2:9091/cockpit/$f154b83f107dfb45e7f4b78ac45b1e96cdc829fcc1b1f7723a442eba73fdab9c/shell/index.js', 'lineNumber': 68, 'columnNumber': 160544}, {'functionName': '', 'scriptId': '21', 'url': 'http://127.0.0.2:9091/cockpit/$f154b83f107dfb45e7f4b78ac45b1e96cdc829fcc1b1f7723a442eba73fdab9c/shell/index.js', 'lineNumber': 68, 'columnNumber': 160548}]}}
error: Modal: Specify at least one of: title, aria-label, aria-labelledby.
error: Modal: Specify at least one of: title, aria-label, aria-labelledby.
cdp: {'source': 'network', 'level': 'error', 'text': 'Failed to load resource: the server responded with a status of 404 (Not Found)', 'url': 'http://127.0.0.2:9091/cockpit/$f154b83f107dfb45e7f4b78ac45b1e96cdc829fcc1b1f7723a442eba73fdab9c/podman/po.js', 'networkRequestId': '3130.66'}
cdp: {'source': 'network', 'level': 'error', 'text': 'Failed to load resource: the server responded with a status of 404 (Not Found)', 'url': 'http://127.0.0.2:9091/cockpit/$f154b83f107dfb45e7f4b78ac45b1e96cdc829fcc1b1f7723a442eba73fdab9c/podman/po.js', 'networkRequestId': '3130.71'}
warning: grep: /sys/class/dmi/id/power/autosuspend_delay_ms: Input/output error

error: Modal: Specify at least one of: title, aria-label, aria-labelledby.
Wrote screenshot to TestApplication-testDownloadImage-rhel-9-0-127.0.0.2-2201-FAIL.png
Wrote HTML dump to TestApplication-testDownloadImage-rhel-9-0-127.0.0.2-2201-FAIL.html
Wrote JS log to TestApplication-testDownloadImage-rhel-9-0-127.0.0.2-2201-FAIL.js.log
Journal extracted to TestApplication-testDownloadImage-rhel-9-0-127.0.0.2-2201-FAIL.log.gz
Warning: Stopping podman.service, but it can still be activated by:
  podman.socket
> warning: transport closed: disconnected
Failed to get session path: No session '19' known
Traceback (most recent call last):
  File "/work/bots/make-checkout-workdir/test/check-application", line 767, in testDownloadImage
    dialog0.openDialog() \
  File "/work/bots/make-checkout-workdir/test/check-application", line 707, in expectDownloadSuccess
    b.wait_visible('#containers-images td[data-label="Name"]:contains("{0}")'.format(self.imageName))
  File "/work/bots/make-checkout-workdir/test/common/testlib.py", line 411, in wait_visible
    self._wait_present(selector)
  File "/work/bots/make-checkout-workdir/test/common/testlib.py", line 402, in _wait_present
    self.wait_js_func('ph_is_present', selector)
  File "/work/bots/make-checkout-workdir/test/common/testlib.py", line 396, in wait_js_func
    self.wait_js_cond("%s(%s)" % (func, ','.join(map(jsquote, args))))
  File "/work/bots/make-checkout-workdir/test/common/testlib.py", line 393, in wait_js_cond
    self.raise_cdp_exception("timeout\nwait_js_cond", cond, result["exceptionDetails"], trailer)
  File "/work/bots/make-checkout-workdir/test/common/testlib.py", line 187, in raise_cdp_exception
    raise Error("%s(%s): %s" % (func, arg, msg))
testlib.Error: timeout
wait_js_cond(ph_is_present("#containers-images td[data-label=\"Name\"]:contains(\"my-busybox\")")): {'exceptionId': 1, 'text': 'Uncaught (in promise) Error: condition did not become true', 'lineNumber': 238, 'columnNumber': 9, 'scriptId': '3', 'stackTrace': {'callFrames': [{'functionName': 'PhWaitCondTimeout', 'scriptId': '3', 'url': '', 'lineNumber': 237, 'columnNumber': 8}, {'functionName': '', 'scriptId': '3', 'url': '', 'lineNumber': 249, 'columnNumber': 23}]}, 'exception': {'type': 'object', 'subtype': 'error', 'className': 'PhWaitCondTimeout', 'description': 'Error: condition did not become true\n    at <anonymous>:250:24', 'objectId': '9128935592755241297.6.2'}}
cdp: {'source': 'network', 'level': 'error', 'text': 'Failed to load resource: the server responded with a status of 401 (Authentication failed)', 'timestamp': 1625043787165.48, 'url': 'http://127.0.0.2:9091/cockpit/login', 'networkRequestId': '3130.12'}
cdp: {'source': 'security', 'level': 'error', 'text': 'Refused to apply inline style because it violates the following Content Security Policy directive: "default-src \'self\' http://127.0.0.2:9091". Either the \'unsafe-inline\' keyword, a hash (\'sha256-BMIkowAqVEI88ibUJ1NtYB8ByZOuS4MTi73eGoFo2+Q=\'), or a nonce (\'nonce-...\') is required to enable inline execution. Note also that \'style-src\' was not explicitly set, so \'default-src\' is used as a fallback.\n', 'url': 'http://127.0.0.2:9091/cockpit/$f154b83f107dfb45e7f4b78ac45b1e96cdc829fcc1b1f7723a442eba73fdab9c/shell/index.js', 'lineNumber': 30, 'stackTrace': {'callFrames': [{'functionName': 'ke', 'scriptId': '21', 'url': 'http://127.0.0.2:9091/cockpit/$f154b83f107dfb45e7f4b78ac45b1e96cdc829fcc1b1f7723a442eba73fdab9c/shell/index.js', 'lineNumber': 30, 'columnNumber': 35691}, {'functionName': 'Ze', 'scriptId': '21', 'url': 'http://127.0.0.2:9091/cockpit/$f154b83f107dfb45e7f4b78ac45b1e96cdc829fcc1b1f7723a442eba73fdab9c/shell/index.js', 'lineNumber': 30, 'columnNumber': 44728}, {'functionName': 'append', 'scriptId': '21', 'url': 'http://127.0.0.2:9091/cockpit/$f154b83f107dfb45e7f4b78ac45b1e96cdc829fcc1b1f7723a442eba73fdab9c/shell/index.js', 'lineNumber': 30, 'columnNumber': 46409}, {'functionName': 'D', 'scriptId': '21', 'url': 'http://127.0.0.2:9091/cockpit/$f154b83f107dfb45e7f4b78ac45b1e96cdc829fcc1b1f7723a442eba73fdab9c/shell/index.js', 'lineNumber': 63, 'columnNumber': 24932}, {'functionName': '', 'scriptId': '21', 'url': 'http://127.0.0.2:9091/cockpit/$f154b83f107dfb45e7f4b78ac45b1e96cdc829fcc1b1f7723a442eba73fdab9c/shell/index.js', 'lineNumber': 63, 'columnNumber': 34130}, {'functionName': '', 'scriptId': '21', 'url': 'http://127.0.0.2:9091/cockpit/$f154b83f107dfb45e7f4b78ac45b1e96cdc829fcc1b1f7723a442eba73fdab9c/shell/index.js', 'lineNumber': 68, 'columnNumber': 160544}, {'functionName': '', 'scriptId': '21', 'url': 'http://127.0.0.2:9091/cockpit/$f154b83f107dfb45e7f4b78ac45b1e96cdc829fcc1b1f7723a442eba73fdab9c/shell/index.js', 'lineNumber': 68, 'columnNumber': 160548}]}}
error: Modal: Specify at least one of: title, aria-label, aria-labelledby.
error: Modal: Specify at least one of: title, aria-label, aria-labelledby.
cdp: {'source': 'network', 'level': 'error', 'text': 'Failed to load resource: the server responded with a status of 404 (Not Found)', 'url': 'http://127.0.0.2:9091/cockpit/$f154b83f107dfb45e7f4b78ac45b1e96cdc829fcc1b1f7723a442eba73fdab9c/podman/po.js', 'networkRequestId': '3130.66'}
cdp: {'source': 'network', 'level': 'error', 'text': 'Failed to load resource: the server responded with a status of 404 (Not Found)', 'url': 'http://127.0.0.2:9091/cockpit/$f154b83f107dfb45e7f4b78ac45b1e96cdc829fcc1b1f7723a442eba73fdab9c/podman/po.js', 'networkRequestId': '3130.71'}
warning: grep: /sys/class/dmi/id/power/autosuspend_delay_ms: Input/output error

error: Modal: Specify at least one of: title, aria-label, aria-labelledby.

# Result testDownloadImage (__main__.TestApplication) failed
# 1 TEST FAILED [70s on 2-cockpit-7]
not ok 5 /work/bots/make-checkout-workdir/test/check-application TestApplication.testDownloadImage [ND@0]
```
First occurrence: 2021-06-30T09:04:12.755855 | revision 2f99caaeb2d578ff52a309b0347d0153390b366f
Times recorded: 1
Latest occurrences:

- 2021-06-30T09:04:12.755855 | revision 2f99caaeb2d578ff52a309b0347d0153390b366f

**Comment by martinpitt** [comment]
> Is it something that happens from time to time?

Apparently -- I only saw it once. If it doesn't come back, it may not be new either, and we can handle it outside of this refresh.

**Review by KKoukiou — APPROVED** [review]

File: naughty/rhel-9/2144-podman-api-call-loses-data (hunk 1 above)
Comment (KKoukiou) [reply naughty/rhel-9/2144-podman-api-call-loses-data:1]: thanks

**Comment by KKoukiou** [comment]
> Do you want to include a naughty for the nft selinux issue as well, or want me to?

I thought that we don't need to include it / report it unless it happens again. I did not see it neither in the last run actually

**Review by martinpitt — APPROVED** [review]
Danke!

**Pull request merged**